test/checkout.int.t.sol

```diff
@@ -1039,13 +946,157 @@ contract Checkout_swapWithoutFeeSpecialOrder is Checkout {
+        amount: 1 ether,
+        vintage: 2020
+      })
+    );
+    _removalIds.push(
+      _seedAndListRemoval({
+        supplier: _namedAccounts.supplier3,
+        amount: 1 ether,
+        vintage: 2019
+      })
+    );
+    uint256 purchaseAmount = _market.calculateCheckoutTotalWithoutFee({
+      amount: certificateAmount,
+      priceMultiple: _priceMultiple
+    });
+    _market.grantRole({role: _market.MARKET_ADMIN_ROLE(), account: owner});
+    vm.prank(_namedAccounts.admin);
+    _bpNori.deposit(owner, abi.encode(purchaseAmount));
+    vm.prank(owner);
+    _bpNori.approve(address(_market), purchaseAmount);
+  }
+
+  function test_basicFulfillment() external {
+    vm.prank(owner);
+    vm.recordLogs();
+    _market.swapWithoutFeeSpecialOrder(
+      owner,
+      owner,
+      certificateAmount,
+      customFee,
+      _priceMultiple,
+      address(0),
+      vintages
+    );
+
+    Vm.Log[] memory entries = vm.getRecordedLogs();
+    // In this test scenario, the `CreateCertificate` event index is 18 because multiple suppliers are getting
+    // paid so we use the event selector to find the event index.
+    uint256 createCertificateEventIndex;
+    for (uint256 i = 0; i < entries.length; ++i) {
+      if (entries[i].topics[0] == CREATE_CERTIFICATE_EVENT_SELECTOR) {
+        createCertificateEventIndex = i;
+        break;
+      }
+    }
+    assertEq(
+      entries[createCertificateEventIndex].topics[0],
+      CREATE_CERTIFICATE_EVENT_SELECTOR
+    );
+    assertEq(
+      entries[createCertificateEventIndex].topics[1],
+      bytes32(uint256(uint160(address(owner))))
+    );
+    assertEq(
+      entries[createCertificateEventIndex].topics[2],
+      bytes32(uint256(uint256(0)))
+    );
+    assertEq(
+      entries[createCertificateEventIndex].topics[3],
+      bytes32(uint256(uint160(address(_bpNori))))
+    );
+    (
+      address from,
+      uint256 eventCertificateAmount,
+      uint256[] memory removalIds,
+      uint256[] memory removalAmounts,
+      uint256 priceMultiple,
+      uint256 noriFeePercentage
+    ) = abi.decode(
+        entries[createCertificateEventIndex].data,
+        (address, uint256, uint256[], uint256[], uint256, uint256)
+      );
+    assertEq(from, address(_removal));
+    assertEq(eventCertificateAmount, certificateAmount);
+    assertEq(priceMultiple, _priceMultiple);
+    assertEq(noriFeePercentage, customFee);
+    assertEq(removalIds.length, 3);
+    assertEq(removalAmounts.length, 3);
+    assertEq(removalIds[0], _removalIds[2]);
+    assertEq(removalIds[1], _removalIds[3]);
+    assertEq(removalIds[2], _removalIds[4]);
+    assertEq(removalAmounts[0], 1 ether);
+    assertEq(removalAmounts[1], 1 ether);
+    assertEq(removalAmounts[2], 0.5 ether);
+  }
+
+  function test_revertsWhenNoRemovalsFromSpecifiedVintages() external {
+    uint256[] memory nonexistentVintages = new uint256[](2);
+    vintages[0] = 2023; // no such vintages available in market
```

Comment:
Nit: Consider using a safer vintage (e.g., 1,9999, etc) that won't accidentally be listed (e.g. with an amount that partially can fill the order thus creating the same error but no test failure)